### ISSUE TITLE
Problems with NOAA NDFD data

### ISSUE THREAD

**Issue by dopplershift** [ecmwf/cfgrib#295]
When opening some NDFD (National Digital Forecast Database) data from NOAA like:
```python
from urllib.request import urlopen

import matplotlib.pyplot as plt
import xarray as xr

url = 'https://noaa-ndfd-pds.s3.amazonaws.com/opnl/AR.conus/VP.001-003/ds.temp.bin'
fname = 'forecast.bin'
with open(fname, 'wb') as cache_file:
    cache_file.write(urlopen(url).read())

ds = xr.open_dataset(fname, engine='cfgrib')
plt.imshow(ds.t2m[0])
```
The result is:
<img width="378" alt="image" src="https://user-images.githubusercontent.com/221526/160027916-308b437b-2a7e-47e6-8d1c-dba7c6b2d491.png">

This contrasts with the same message opened with `pygrib`:
```python
import pygrib
grb = pygrib.open(fname)
msg = grb.message(1)
plt.imshow(msg.values)
```
<img width="374" alt="image" src="https://user-images.githubusercontent.com/221526/160028747-5a219d96-2469-449e-8dea-f9508afb38c3.png">

This seems to be caused by the fact that the "alternativeRowScanning" key is not considered at all by cfgrib, and it is set for this data:
```python
>>> print(msg['alternativeRowScanning'])
1
```

**Comment by shahramn** [comment]
Which version of the ecCodes library are you using?
I just tried the Geoiterator on the first message of the file ds.temp.bin and it looks fine ( I am using ecCodes v2.25.0 )

**Comment by dopplershift** [comment]
That's interesting. I have eccodes 2.25.0 installed, and I still had problems when going through cfgrib. I'll continue the discussion over on #296 .

**Comment by iainrussell** [comment]
Fixed with #296
